Sources/Datadog/Core/System/NetworkConnectionInfoProvider.swift

```diff
@@ -40,12 +40,16 @@ internal protocol NetworkConnectionInfoProviderType {
 }
 
 internal class NetworkConnectionInfoProvider: NetworkConnectionInfoProviderType {
-    private let queue: DispatchQueue?
-    private let fetchBlock: () -> NetworkConnectionInfo
-    private let cancelBlock: (() -> Void)?
+    /// Queue synchronizing the reads and updates to `NetworkConnectionInfo`.
+    private let queue = DispatchQueue(
+        label: "com.datadoghq.network-connection-info",
+        target: .global(qos: .utility)
+    )
+    /// The `NetworkConnectionInfoProviderType` depending on the OS version.
+    private let wrappedProvider: NetworkConnectionInfoProviderType
 
     var current: NetworkConnectionInfo {
-        return fetchBlock()
+        queue.sync { wrappedProvider.current }
```

Comment:
I just saw that after writing the first comment :)

Yes, this should ensure thread safety 👍 

But if we use `pathUpdateHandler` to read `currentPath`, we can 
1. remove `queue` property from this class
2. reduce amount of `queue.sync {...}` calls
3. make thread safety the responsibility of `NWPathMonitor`; regardless of the queue we provide, we will be reading from the write queue